xarray/core/duck_array_ops.py

```diff
@@ -72,7 +72,7 @@ def fail_on_dask_array_input(values, msg=None, func_name=None):
 
 
 if hasattr(np, "isnat") and (
-    dask_array is None or hasattr(dask_array_type, "__array_ufunc__")
+    dask_array_type == () or hasattr(dask_array_type[0], "__array_ufunc__")
```

Comment:
not sure about this, does anyone have a better idea?

Reply:
can you describe the problem please?

Reply:
the issue is that `dask_array_type` is always a tuple: https://github.com/pydata/xarray/blob/3ef75aeffaf89aa1c7ea9957e5bdf8d1b36aac54/xarray/core/pycompat.py#L5-L11
so for checking if `dask.array.Array` implements `__array_function__` we need to use its first element

Reply:
wait we are now numpy >= 0.15 and dask >= 2.2. Can we get rid of the if condition?

Reply:
I think so? `__array_ufunc__` is supported since `numpy>=0.13` and `dask>=0.15`.

Reply:
I replaced the condition with a call to `_dask_or_eager_func`.